### PR TITLE
99 storybook

### LINKED ISSUE #99: Storybook
**Story**
Install storybook and add the required configuration with a simple example

### PR DESCRIPTION
**Pull request by elhaw**
resolves #99

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  framework: '@storybook/react',
+  core: {
+    builder: '@storybook/builder-webpack5',
+  },
+  staticDirs: ['../public'],
+  webpackFinal: async (config) => {
+    config.module.rules.push({
+      test: /\.css$/,
+      use: ['postcss-loader'],
+    });
```
Comment (atwfk) [inline .storybook/main.js]: remove `postcss-loader` and install `@storybook/addon-postcss` instead,
and add `storybook-css-modules-preset` for loading css modules to storybook

File: package.json [hunk 2]
```diff
@@ -48,6 +60,7 @@
     "@typescript-eslint/eslint-plugin": "^5.25.0",
     "@typescript-eslint/parser": "^5.25.0",
     "autoprefixer": "^10.4.7",
+    "babel-loader": "^8.2.5",
```
Comment (atwfk) [inline package.json:65]: why do we need to install `babel-loader` ?, storybook works fine for me without installing it

**Review by atwfk — COMMENTED** [review]
format storybook configuration files `main.js` and `preview.js` to fix prettier check

File: package.json (hunk 2 above)
Comment (elhaw) [reply package.json:65]: All the packages has been added automatically after installing storybook using this command

> npx storybook init
[installation guide](https://storybook.js.org/docs/react/get-started/install)

File: .storybook/main.js (hunk 1 above)
Comment (elhaw) [reply .storybook/main.js]: After installing them I got this error even after upgrading the packages according to the guide in this link https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users. 



> yarn run v1.22.19
> $ start-storybook -p 6006
> info @storybook/react v6.5.12
> info
> info => Loading presets
> info => Serving static files from ././public at /
> info Addon-docs: using MDX1
> info => Using PostCSS preset with postcss@7.0.39
> info => Using default Webpack5 setup
> <i> [webpack-dev-middleware] wait until bundle finished
> 10% building 0/1 entries 0/0 dependencies 0/0 modules
> info => Using cached manager
> 99% done plugins webpack-hot-middlewarewebpack built preview d858ecb933a32b257de5 in 4908ms
> ModuleBuildError: Module build failed (from ./node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/cjs.js):
> Error: PostCSS plugin tailwindcss requires PostCSS 8.
> Migration guide for end-users:
> https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
>     at Processor.normalize (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss/lib/processor.js:153:15)
>     at new Processor (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss/lib/processor.js:56:25)
>     at postcss (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss/lib/postcss.js:55:10)
>     at Object.loader (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/index.js:95:20)
>     at processResult (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/webpack/lib/NormalModule.js:758:19)
>     at /Users/elhaw/projects/m3ntorship.com-v2/node_modules/webpack/lib/NormalModule.js:860:5
>     at /Users/elhaw/projects/m3ntorship.com-v2/node_modules/loader-runner/lib/LoaderRunner.js:400:11
>     at /Users/elhaw/projects/m3ntorship.com-v2/node_modules/loader-runner/lib/LoaderRunner.js:252:18
>     at context.callback (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
>     at Object.loader (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/index.js:104:7)
> ModuleBuildError: Module build failed (from ./node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/cjs.js):
> Error: PostCSS plugin tailwindcss requires PostCSS 8.
> Migration guide for end-users:
> https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
>     at Processor.normalize (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss/lib/processor.js:153:15)
>     at new Processor (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss/lib/processor.js:56:25)
>     at postcss (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss/lib/postcss.js:55:10)
>     at Object.loader (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/index.js:95:20)
>     at processResult (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/webpack/lib/NormalModule.js:758:19)
>     at /Users/elhaw/projects/m3ntorship.com-v2/node_modules/webpack/lib/NormalModule.js:860:5
>     at /Users/elhaw/projects/m3ntorship.com-v2/node_modules/loader-runner/lib/LoaderRunner.js:400:11
>     at /Users/elhaw/projects/m3ntorship.com-v2/node_modules/loader-runner/lib/LoaderRunner.js:252:18
>     at context.callback (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
>     at Object.loader (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/index.js:104:7)
> ModuleBuildError: Module build failed (from ./node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/cjs.js):
> Error: PostCSS plugin tailwindcss requires PostCSS 8.
> Migration guide for end-users:
> https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
>     at Processor.normalize (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss/lib/processor.js:153:15)
>     at new Processor (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss/lib/processor.js:56:25)
>     at postcss (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss/lib/postcss.js:55:10)
>     at Object.loader (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/index.js:95:20)
>     at processResult (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/webpack/lib/NormalModule.js:758:19)
>     at /Users/elhaw/projects/m3ntorship.com-v2/node_modules/webpack/lib/NormalModule.js:860:5
>     at /Users/elhaw/projects/m3ntorship.com-v2/node_modules/loader-runner/lib/LoaderRunner.js:400:11
>     at /Users/elhaw/projects/m3ntorship.com-v2/node_modules/loader-runner/lib/LoaderRunner.js:252:18
>     at context.callback (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
>     at Object.loader (/Users/elhaw/projects/m3ntorship.com-v2/node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/index.js:104:7)
> 
> WARN Broken build, fix the error above.
> WARN You may need to refresh the browser.
> 
> info => Loading presets
> 99% done plugins webpack-hot-middlewarewebpack built preview 3406cdf3120fabc87de5 in 147ms
>

Comment (atwfk) [reply .storybook/main.js]: I did these steps and it worked fine for me:
- `yarn remove postcss-loader && yarn add -D @storybook/addon-postcss storybook-css-modules-preset`
-  in `main.js` file, removed `postcss-loader` config 
```javascript
    config.module.rules.push({
      test: /\.css$/,
      use: ['postcss-loader'],
    });
```
- and added the configs for the installed packages
```javascript
  addons: [
    ...,
    'storybook-css-modules-preset',
    {
      name: '@storybook/addon-postcss',
      options: {
        postcssLoaderOptions: {
          implementation: require('postcss'),
        },
      },
    },
  ],
```

**Review by atwfk — APPROVED** [review]
LGTM

**Pull request merged**